### DISCUSSION TITLE
Does this project have any utility for working with binary files?

### DISCUSSION THREAD

**Discussion by colbyn** [arxanas/git-branchless#600] [Q&A]
I have a very general question if you don’t mind: in summary, does this project have any utility for working with binary files? 

Perhaps this is out of the scope of this project, but I thought I'd ask just in case. 

TLDR

I’d like to be able to save and update app generated binary files in a git repo, and do so as efficiently as possible (to stay under GitHub’s free tier size limits).

-To elaborate-

I’m working on a [note-taking project](https://github.com/subscript-publishing/subscript) (notably for school notes) that supports a markup language that can include ‘drawing’, essentially binary files from the iPad freeform drawing apps, and I’d like to store both the plain-text markup files and binary drawing files in the git tree, and as I mentioned, do so as efficiently as possible to stay under GitHub’s free tier size limits. Furthermore, synchronizing files generated VIA the iPad and accessing such on my laptop is a huge pain, and I’m kinda thinking about setting up the iPad apps to work with git repos directly in some way.

**Comment by arxanas** [comment]
Not in particular. If you need to manipulate commits in-memory instead, then this project might be for you.

You can explore using Git LFS or implementing your own clean/smudge filters to handle binary asset management.
